examples/scratch-env/main.go

```diff
@@ -68,39 +61,47 @@ func runMain() int {
 	cfg, err := env.Start()
 	if err != nil {
 		log.Error(err, "unable to start the test environment")
+		// shut down the environment in case we started it and failed while
+		// installing CRDs or provisioning users.
+		if err := env.Stop(); err != nil {
+			log.Error(err, "unable to stop the test environment after an error (this might be expected, but just though you should know)")
+		}
 		return 1
 	}
 
 	log.Info("apiserver running", "host", cfg.Host)
 
-	// TODO(directxman12): add support for writing to a new context in an existing file
-	kubeconfigFile, err := ioutil.TempFile("", "scratch-env-kubeconfig-")
+	// NB(directxman12): this group is unfortunately named, but various
+	// kubernetes versions require us to use it to get "admin" access.
+	user, err := env.ControlPlane.AddUser(envtest.User{
+		Name:   "envtest-admin",
+		Groups: []string{"system:masters"},
+	}, nil)
 	if err != nil {
-		log.Error(err, "unable to create kubeconfig file, continuing on without it")
+		log.Error(err, "unable to provision admin user, continuing on without it")
```

Comment:
This effectively makes it unusable, shouldn't we return an error (perhaps a well known one in case ppl want to check for it?)

Reply:
ah, yeah, I think that was an artifact of the way that control flow worked before.  Good point :+1: